src/lib/dlt_user.c

```diff
@@ -504,7 +512,22 @@ DltReturnValue dlt_init(void)
                   " Shared memory %s cannot be created!\n", dltShmName);
 
 #endif
+#ifdef DLT_TRACE_LOAD_CTRL_ENABLE
+    /* initialize for trace load check */
+
+    pthread_rwlock_init(&trace_load_rw_lock, NULL);
+    pthread_rwlock_wrlock(&trace_load_rw_lock);
+
+    trace_load_settings = malloc(sizeof(DltTraceLoadSettings));
```

Comment:
Kindly please ensure that the memory is allocated successfully before accessing.

Reply:
done